pkg-manager/plugin-commands-installation/src/dedupe.ts

```diff
@@ -1,14 +1,15 @@
 import { docsUrl } from '@pnpm/cli-utils'
-import { UNIVERSAL_OPTIONS } from '@pnpm/common-cli-options-help'
+import { OPTIONS, UNIVERSAL_OPTIONS } from '@pnpm/common-cli-options-help'
 import { dedupeDiffCheck } from '@pnpm/dedupe.check'
 import renderHelp from 'render-help'
-import { type InstallCommandOptions } from './install'
+import { type InstallCommandOptions, rcOptionsTypes as installCommandRcOptionsTypes } from './install'
 import { installDeps } from './installDeps'
-import { types as allTypes } from '@pnpm/config'
-import pick from 'ramda/src/pick'
+import omit from 'ramda/src/omit'
 
+// In general, the "pnpm dedupe" command should use .npmrc options that "pnpm install" would also accept.
 export function rcOptionsTypes () {
-  return pick(['ignore-scripts'], allTypes)
+  // Some options on pnpm install (like --frozen-lockfile) don't make sense on pnpm dedupe.
+  return omit(['frozen-lockfile'], installCommandRcOptionsTypes())
 }
```

Comment:
This makes `rcOptionTypes()` more similar to `pnpm deploy`.

https://github.com/pnpm/pnpm/blob/693944b66994f50f5f1a940e10e0f0dd0c98dae2/releasing/plugin-commands-deploy/src/deploy.ts#L17-L23